dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -323,6 +370,113 @@ def _update_stop_condition(self, data: DataFrame):
+            cms.update(value)
+            i_count = cms.get_estimate(value)
+            i_total_count = i_count + self.cms.get_estimate(value)
+            # approximate heavy-hitters
+            if i_count >= int(len_df / self._cms_max_num_heavy_hitters):
+                heavy_hitter_dict[value] = i_count
+                # missing_heavy_hitter_dict.pop(value, None)
+            elif i_total_count >= int(
+                (self.sample_size + len_df) / self._cms_max_num_heavy_hitters
+            ):
+                missing_heavy_hitter_dict[value] = i_total_count
+                # heavy_hitter_dict.pop(value, None)
+
+        return cms, heavy_hitter_dict, missing_heavy_hitter_dict
+
+    @BaseColumnProfiler._timeit(name="categories")
+    def _merge_categories_cms(
+        self,
+        cms1,
+        heavy_hitter_dict1,
+        len1,
+        missing_heavy_hitter_dict,
+        cms2,
+        heavy_hitter_dict2,
+        len2,
+        other_max_num_heavy_hitters,
+    ):
+        """Return the aggregate count min sketch and approximate histogram (categories).
+
+        :param cms1: count min sketch
+        :type cms1: datasketches.countminsketch
+        :param cms2: count min sketch
+        :type cms2: datasketches.countminsketch
+        :param heavy_hitter_dict1: Heavy Hitters category count
+        :type heavy_hitter_dict1: Dict
+        :param heavy_hitter_dict2: Heavy Hitters category count
+        :type heavy_hitter_dict2: Dict
+        :param missing_heavy_hitter_dict2: Heavy Hitters category count
+        considering two batches are two chunks of a data stream
+        :type missing_heavy_hitter_dict2: Dict
+        :param len1: number of samples in batch 1
+        :type len1: int
+        :param len2: number of samples in batch 2
+        :type len2: int
+        :return: cms1, categories
+        """
+        try:
+            cms3 = datasketches.count_min_sketch(
+                self.cms_num_hashes, self.cms_num_buckets
+            )
+            cms3.merge(cms1)
+            cms3.merge(cms2)
+        except ValueError as err:
+            raise err(
+                """Incompatible sketch configuration. When merging two sketches,
+                they must have the same number of buckets and hashes,
+                which are defined by cms_confidence and cms_relative_error options,
+                respectively."""
+            )
+
+        # re-collecting the estimates of non intersecting categories before
+        # re-applying heavy-hitters to the aggregate profile.
+        missing_in_1 = [
+            k for k in heavy_hitter_dict1.keys() if k not in heavy_hitter_dict2.keys()
+        ]
+        for k in missing_in_1:
+            heavy_hitter_dict2[k] = cms2.get_estimate(k)
+        missing_in_2 = [
+            k for k in heavy_hitter_dict2.keys() if k not in heavy_hitter_dict1.keys()
+        ]
+        for k in missing_in_2:
+            heavy_hitter_dict1[k] = cms1.get_estimate(k)
```

Comment:
good call on this for the streaming case to catch.
I think we can clean this by doing:
```python
# re-collecting the estimates of non intersecting categories before
# re-applying heavy-hitters to the aggregate profile.
heavy_hitter_dict1 = heavy_hitter_dict1.copy()
heavy_hitter_dict2 = heavy_hitter_dict2.copy()
for k in (x for x in heavy_hitter_dict1 if x not in heavy_hitter_dict2):
    heavy_hitter_dict2[k] = cms2.get_estimate(k)
for k in (x for x in heavy_hitter_dict2 if x not in heavy_hitter_dict1):
    heavy_hitter_dict1[k] = cms1.get_estimate(k)
```

now down to a single loop and we don't have to call keys to loop.

Additionally, since this is now altering the dict, we need to copy any dict being passed in here so we don't alter the original.

Reply:
anything passing in `self` or `other` for the heavy_hitter_dict would mutate the original dict otherwise.

Reply:
nice!